wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/ShareableDebugReportView.swift

```diff
@@ -0,0 +1,127 @@
+import WireDataModel
+import WireDesign
+
+class ShareableDebugReportView: UIView {
+
+    // MARK: - Views
+
+    private let topLabel: UILabel = {
+        let topLabel = UILabel()
+        topLabel.numberOfLines = 1
+        topLabel.lineBreakMode = .byTruncatingMiddle
+        return topLabel
+    }()
+
+    private let bottomLabel: UILabel = {
+        let bottomLabel = UILabel()
+        bottomLabel.numberOfLines = 1
+        return bottomLabel
+    }()
+
+    private let labelsView = UIView()
+
+    private let documentIconView: UIImageView = {
+        let documentIconView = UIImageView()
+        documentIconView.tintColor = SemanticColors.Icon.backgroundDefault
+        documentIconView.contentMode = .center
+        documentIconView.setTemplateIcon(.document, size: .small)
+        return documentIconView
+    }()
+
+    // MARK: - Life cycle
+
+    convenience init() {
+        self.init(frame: .zero)
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupViews()
+        createConstraints()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+
+    private func setupViews() {
+        addSubview(labelsView)
+        addSubview(documentIconView)
+        labelsView.addSubview(topLabel)
+        labelsView.addSubview(bottomLabel)
+    }
+
+    private func createConstraints() {
+        [
+            labelsView,
+            topLabel,
+            bottomLabel,
+            documentIconView
+        ].forEach { $0.translatesAutoresizingMaskIntoConstraints = false }
+
+        NSLayoutConstraint.activate([
+            topLabel.topAnchor.constraint(equalTo: labelsView.topAnchor),
+            topLabel.leftAnchor.constraint(equalTo: labelsView.leftAnchor),
+            topLabel.rightAnchor.constraint(equalTo: labelsView.rightAnchor),
+
+            bottomLabel.topAnchor.constraint(equalTo: topLabel.bottomAnchor, constant: 2),
+            bottomLabel.leftAnchor.constraint(equalTo: labelsView.leftAnchor),
+            bottomLabel.rightAnchor.constraint(equalTo: labelsView.rightAnchor),
+            bottomLabel.bottomAnchor.constraint(equalTo: labelsView.bottomAnchor),
+
+            labelsView.leftAnchor.constraint(equalTo: documentIconView.rightAnchor, constant: 12),
+            labelsView.rightAnchor.constraint(equalTo: rightAnchor, constant: -12),
+            labelsView.centerYAnchor.constraint(equalTo: centerYAnchor),
+
+            documentIconView.topAnchor.constraint(equalTo: topAnchor, constant: 12),
+            documentIconView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -12),
+            documentIconView.widthAnchor.constraint(equalToConstant: 32),
+            documentIconView.heightAnchor.constraint(equalToConstant: 32),
+            documentIconView.centerYAnchor.constraint(equalTo: centerYAnchor),
+            documentIconView.leftAnchor.constraint(equalTo: leftAnchor, constant: 12)
```

Comment:
suggestion: Create an enum to hold all of those numbers, so we can avoid using magic numbers. 

```swift
enum LayoutConstants {
    static let spacing: CGFloat = 2
    static let padding: CGFloat = 12
    static let iconSize: CGFloat = 32
}

NSLayoutConstraint.activate([
    topLabel.topAnchor.constraint(equalTo: labelsView.topAnchor),
    topLabel.leftAnchor.constraint(equalTo: labelsView.leftAnchor),
    topLabel.rightAnchor.constraint(equalTo: labelsView.rightAnchor),

    bottomLabel.topAnchor.constraint(equalTo: topLabel.bottomAnchor, constant: LayoutConstants.spacing),
    bottomLabel.leftAnchor.constraint(equalTo: labelsView.leftAnchor),
    bottomLabel.rightAnchor.constraint(equalTo: labelsView.rightAnchor),
    bottomLabel.bottomAnchor.constraint(equalTo: labelsView.bottomAnchor),

    labelsView.leftAnchor.constraint(equalTo: documentIconView.rightAnchor, constant: LayoutConstants.padding),
    labelsView.rightAnchor.constraint(equalTo: rightAnchor, constant: -LayoutConstants.padding),
    labelsView.centerYAnchor.constraint(equalTo: centerYAnchor),

    documentIconView.topAnchor.constraint(equalTo: topAnchor, constant: LayoutConstants.padding),
    documentIconView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -LayoutConstants.padding),
    documentIconView.widthAnchor.constraint(equalToConstant: LayoutConstants.iconSize),
    documentIconView.heightAnchor.constraint(equalToConstant: LayoutConstants.iconSize),
    documentIconView.centerYAnchor.constraint(equalTo: centerYAnchor),
    documentIconView.leftAnchor.constraint(equalTo: leftAnchor, constant: LayoutConstants.padding)
])
```

Reply:
done https://github.com/wireapp/wire-ios/pull/1658/commits/8c85d9ddbe3d4f96ae06fb91a9db48865e963819